reposcan/database/cvemap_store.py

```diff
@@ -86,23 +86,33 @@ def _update_cves(self, to_update):
             finally:
                 cur.close()
 
-    def _populate_cves(self, cvemap):
+    def _set_null_source_cves(self, to_delete):
+        if to_delete:
+            cur = self.conn.cursor()
+            try:
+                execute_values(cur, """update cve set source_id = null where id in (%s)""",
+                               to_delete, page_size=len(to_delete))
+                self.conn.commit()
+            except Exception:  # pylint: disable=broad-except
+                self.logger.exception("Failure while deleting CVEs")
+                self.conn.rollback()
+            finally:
+                cur.close()
+
+    def _populate_cves(self, cvemap):  # pylint: disable=too-many-branches
         cve_impact_map = self._populate_cve_impacts()
         rh_source_id = self._get_source_id('Red Hat')
         cur = self.conn.cursor()
 
         cve_data = cvemap.list_cves()
-        cve_names = [(name,) for name in cve_data]
-        execute_values(cur, """select id, name, source_id,
+        cur.execute("""select id, name, source_id,
                                  description, impact_id,
                                  published_date, modified_date,
                                  cvss3_score, cvss3_metrics,
                                  iava, redhat_url,
                                  secondary_url, source_id,
-                                 cvss2_score, cvss2_metrics
-                                 from cve
-                                 join (values %s) t(name)
-                                using (name)""", cve_names, page_size=len(cve_names))
```

Comment:
removing rhis code causes an error when CVE is removed from CVE map and then re-occurs again later:

> vmaas-reposcan        | f82db87bb331 2019-11-25 16:40:46,356 database.cve_common: [ERROR] Failure while importing CVEs'Traceback (most recent call last):\n  File "/reposcan/database/cvemap_store.py", line 48, in _import_cves\n    list(to_import), page_size=len(to_import))\n  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psycopg2/extras.py", line 1281, in execute_values\n    cur.execute(b\'\'.join(parts))\npsycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "cve_name_key"\nDETAIL:  Key (name)=(CVE-1999-0018) already exists.\n'|